projects/batfish-common-protocol/src/test/java/org/batfish/storage/FileBasedStorageTest.java

```diff
@@ -703,175 +699,121 @@ private static void expectFileNotFoundException(IOExceptionThrower r) {
+    SnapshotId snapshotId = new SnapshotId("snapshot-id");
+    AnswerId answerId = new AnswerId("answer-id");
+    Path networkDir = _storage.getNetworkDir(networkId);
+
+    // write a new answer object for the snapshot
+    Path answerDir = _storage.getAnswerDir(networkId, snapshotId, answerId);
+    _storage.mkdirs(answerDir);
+    _storage.writeStringToFile(answerDir.resolve("answer"), "answer", UTF_8);
+
+    // write a new blob
+    Path blobPath = _storage.getNetworkBlobPath(networkId, "key");
+    _storage.storeNetworkBlob(new ByteArrayInputStream("blob".getBytes()), networkId, "key");
+
+    // should not be expunged since we have new data
+    _storage.runGarbageCollection();
+    assertTrue(Files.exists(networkDir));
+
+    Instant oldTime = Instant.now().minus(GC_SKEW_ALLOWANCE).minus(1, ChronoUnit.MINUTES);
+
+    // don't expunge when the blob dir is old and the snapshot is new
+    Files.setLastModifiedTime(blobPath.getParent(), FileTime.from(oldTime));
+    _storage.runGarbageCollection();
+    assertTrue(Files.exists(_storage.getNetworkDir(networkId)));
+
+    // don't expunge when the snapshot dir is old and the blob is new
+    Files.setLastModifiedTime(blobPath.getParent(), FileTime.from(Instant.now()));
+    Files.setLastModifiedTime(_storage.getSnapshotsDir(networkId), FileTime.from(oldTime));
+    Files.setLastModifiedTime(
+        _storage.getSnapshotDir(networkId, snapshotId), FileTime.from(oldTime));
+    _storage.runGarbageCollection();
+    assertTrue(Files.exists(_storage.getNetworkDir(networkId)));
+
+    // expunge when everything is old
+    Files.setLastModifiedTime(blobPath.getParent(), FileTime.from(oldTime));
+    Files.setLastModifiedTime(_storage.getSnapshotsDir(networkId), FileTime.from(oldTime));
+    Files.setLastModifiedTime(
+        _storage.getSnapshotDir(networkId, snapshotId), FileTime.from(oldTime));
+    _storage.runGarbageCollection();
+    assertFalse(Files.exists(_storage.getNetworkDir(networkId)));
+  }
+
+  /** Check that extant snapshots are not expunged even if their content is old */
+  @Test
+  public void testRunGarbageCollection_extantSnapshot() throws IOException {
+    NetworkId networkId = new NetworkId("network-id");
+    SnapshotId snapshotId = new SnapshotId("snapshot-id");
+    AnswerId answerId = new AnswerId("answer-id");
+
+    _storage.writeId(networkId, "network"); // make the network extant
+    _storage.writeId(snapshotId, "snapshot", networkId); // make the snapshot extant
+
+    // write an answer and make the snapshot data old
+    Path answerDir = _storage.getAnswerDir(networkId, snapshotId, answerId);
+    _storage.mkdirs(answerDir);
+    _storage.writeStringToFile(answerDir.resolve("answer"), "answer", UTF_8);
+
+    Instant oldTime = Instant.now().minus(GC_SKEW_ALLOWANCE).minus(1, ChronoUnit.MINUTES);
+    Files.setLastModifiedTime(
+        _storage.getSnapshotOutputDir(networkId, snapshotId), FileTime.from(oldTime));
```

Comment:
To be clear, the only date being checked in this PR is the last write time of the snapshot output directory. This will not be modified by writing anything in a subfolder of the snapshot output directory. Answers, in particular, are in a subfolder. So writing an answer won't affect the snapshot output folder's last write time.
Perhaps more data should be checked recursively to compute this last write time.

As is, this test is not convincing.